### PR TITLE
DLP Embargo UI

### LINKED ISSUE #996: Update DLP to display embargoed dandisets properly
Add UI for displaying embargo state and for unembargoing ("releasing").

[design doc](https://github.com/dandi/dandi-api/blob/96c85ca5b3c1932b450cb00d4c375f0a176e2261/doc/design/embargo-mvp.md?plain=1#L14-L17)

### PR DESCRIPTION
**Pull request by mvandenburgh**
Closes #996.
Screenshot from 2022-01-04 14-22-25

Screenshot from 2022-01-04 14-22-36

### CODE REVIEW

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          Release Dandiset
+        </v-card-title>
+        <v-divider class="my-3" />
+        <v-card-text>
+          <span>
+            This action will unembargo this dandiset. Note that this is a
+          </span>
+          <span class="font-weight-bold">
+            permanent
+          </span>
```
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:23]: Not a huge deal but I'm pretty sure you can nest this "inline" in the nonbold text currently in the `span` elements before and after this one.

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  >
+    <v-dialog
+      v-if="showWarningDialog"
+      v-model="showWarningDialog"
+      persistent
+      max-width="60vh"
+    >
+      <v-card class="pb-3">
+        <v-card-title class="text-h5 font-weight-light">
+          Release Dandiset
```
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:14]: We may want to just use the term "unembargo" here. "Release" and "publish" are close enough in meaning that people will be confused by this.

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        <v-card-text>
+          <span>
+            This action will unembargo this dandiset. Note that this is a
+          </span>
+          <span class="font-weight-bold">
+            permanent
+          </span>
+          <span>
+            action and is not undoable. Once a dandiset has been released,
+            it will be public forever.
```
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:26]: I'd suggest pulling back from this a bit, with

```suggestion
            it cannot be re-embargoed.
```

instead. "Public forever" is not technically true (admins can help to hide a dandiset that has leaked something illegally, for instance, etc.), and I don't want to scare users into starting to believe something that's not true.

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        <v-divider class="my-3" />
+        <v-card-text>
+          <span>
+            This action will unembargo this dandiset. Note that this is a
+          </span>
+          <span class="font-weight-bold">
+            permanent
+          </span>
+          <span>
+            action and is not undoable. Once a dandiset has been released,
```
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:25]: s/released/unembargoed/

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            it will be public forever.
+            <br><br>
+            Note: this may take a while if your dandiset is large.
+          </span>
+        </v-card-text>
+        <v-card-text>
+          Would you like to proceed?
```
Comment (waxlamp) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:32]: This is a case where we should ask the user to take a deliberate action to indicate they know what they're doing. Typing a random phrase in is a good option (Multinet does this), as would be making the user type in the dandiset ID.

**Review by waxlamp — CHANGES_REQUESTED** [review]
This looks great overall. Some suggestions regarding low-level behavior of the feature.

File: src/views/DandisetLandingView/DandisetUnembargo.vue (hunk 2 above)
Comment (dchiquito) [reply src/views/DandisetLandingView/DandisetUnembargo.vue:14]: I switched the design doc and all of my code to say "Unembargo".

File: src/views/DandisetLandingView/DandisetUnembargo.vue [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+    async function release() {
+      if (currentDandiset.value) {
+        // Display the warning dialog before releasing
+        if (!showWarningDialog.value) {
+          showWarningDialog.value = true;
+          return;
+        }
+
+        await dandiRest.publish(currentDandiset.value.dandiset.identifier);
```
Comment (dchiquito) [inline src/views/DandisetLandingView/DandisetUnembargo.vue:131]:
```suggestion
        await dandiRest.unembargo(currentDandiset.value.dandiset.identifier);
```
The `publish` endpoint is not being recycled, there should be a new method just for unembargoing.

**Review by dchiquito — CHANGES_REQUESTED** [review]

File: src/views/FileBrowserView/FileBrowser.vue [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div>
+  <div v-if="!currentDandiset.dandiset.locked">
```
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply src/views/FileBrowserView/FileBrowser.vue:2]: Yeah, the first `dandiset` is the vuex module, the second is the `dandiset` variable in the store (which is actually a `Version`), and the third `dandiset` refers to the `dandiset` property of the `Version`.

Wow. yeah after reading what I just typed, I agree we need to make some changes. I'll file an issue.

[the remark this answers is not in the document]
Comment (mvandenburgh) [reply src/views/FileBrowserView/FileBrowser.vue:2]: #1023

File: src/views/DandisetLandingView/DandisetUnembargo.vue (hunk 1 above)
Comment (mvandenburgh) [reply src/views/DandisetLandingView/DandisetUnembargo.vue]: https://github.com/dandi/dandiarchive/pull/1021/commits/7bfc45f0c26446cbc2ab416bd2c4d8bf6052cf9b

File: src/views/DandisetLandingView/DandisetUnembargo.vue (hunk 2 above)
Comment (mvandenburgh) [reply src/views/DandisetLandingView/DandisetUnembargo.vue]: https://github.com/dandi/dandiarchive/pull/1021/commits/5ef9785d0ccfdede84bd5ee1e7b04da10992cc2a

File: src/views/DandisetLandingView/DandisetUnembargo.vue (hunk 3 above)
Comment (mvandenburgh) [reply src/views/DandisetLandingView/DandisetUnembargo.vue]: https://github.com/dandi/dandiarchive/pull/1021/commits/a9c59551da129aa9c9d26a738b96ed78c6dd0f20

File: src/types/index.ts [hunk 8]
```diff
@@ -17,6 +17,8 @@ export interface Dandiset {
   // eslint-disable-next-line no-use-before-define
   most_recent_published_version?: Version,
   contact_person?: string,
+  embargo_status: 'EMBARGOED' | 'UNEMBARGOING' | 'OPEN',
+  locked: boolean,
```
[the remark this answers is not in the document]
Comment (mvandenburgh) [reply src/types/index.ts]: https://github.com/dandi/dandiarchive/pull/1021/commits/78a50a5aef329b1d180beb08a3cf7d9e7a74c96b

File: src/views/DandisetLandingView/DandisetUnembargo.vue (hunk 6 above)
Comment (mvandenburgh) [reply src/views/DandisetLandingView/DandisetUnembargo.vue]: https://github.com/dandi/dandiarchive/pull/1021/commits/a24250359265894b2a90a0946f3b130c8b6069f9

**Review by waxlamp — APPROVED** [review]

**Pull request merged**